sklearn/metrics/pairwise.py

```diff
@@ -1216,6 +1369,14 @@ def pairwise_distances(X, Y=None, metric="euclidean", n_jobs=1, **kwds):
                          "Valid metrics are %s, or 'precomputed', or a "
                          "callable" % (metric, _VALID_METRICS))
 
+    if metric in _MASKED_SUPPORTED_METRICS:
+        missing_values = kwds.get("missing_values") if kwds.get(
+            "missing_values") is not None else np.nan
```

Comment:
Why don't we just do this in masked_euclidean_distances. I don't think we benefit from having it here.

Reply:
Well I figured it would be handy when other masked_metrics get added (in fact I plan to add a masked_manhattan once this is merged). Additionally, masked_euclidean_distances is also not passed the kwds dictionary in current API. So unless you think changing this is important, I would prefer to leave this as is. What do you think?